utam-compiler/src/main/java/utam/compiler/translator/Manifest.java

```diff
@@ -0,0 +1,76 @@
+package utam.compiler.translator;
+
+import static utam.core.framework.UtamLogger.info;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/** Writes manifest of generated files */
+public class Manifest {
+  private final List<String> resourcePaths = new ArrayList<>();
+
+  /**
+   * Writes the path of a geneated resource to the manifest
+   *
+   * @param resourcePath the path of the generated resource
+   */
+  void addWrittenResource(String resourcePath) {
+    resourcePaths.add(resourcePath);
+  }
+
+  /**
+   * Writes the manifest to the specified path
+   *
+   * @param manifestPath the path to write the manifest to
+   */
+  void writeToPath(String manifestPath) {
+    info("write artifact manifest " + manifestPath);
+    try {
+      Collections.sort(resourcePaths);
+      Writer writer = new FileWriter(manifestPath);
+      writeManifest(writer);
+    } catch (IOException e) {
+      String err = String.format("error creating manifest %s", manifestPath);
```

Comment:
nit: our usual practice is to use constant for static text

Reply:
Done in this instance, but pattern is not enforced everywhere.